commands/poll.js

```diff
@@ -0,0 +1,151 @@
+						if (m.content.toLowerCase() === 'none') {
+							message.channel.send(`No worries, I won't add a description to this poll.`);
+							poll_options['description'] = '';
+						} else {
+							message.channel.send(`Sure, the description will be:\n"${m.content}"`);
+							poll_options['description'] = m.content;
+						}
+						stage = 2;
+						poll_options['options'] = {};
+						message.channel.send("**Finally**, add some options to the poll. Send me up to 20 messages starting with the emoji to use at the start, or leave off an emoji to use letters.\ne.g. `:egg: Option with custom emoji`, or `Option with default emoji`\nWhen you're done, type `done`.");
+					} else if (stage == 2) { // Options
+						let done = false;
+						if (m.content.toLowerCase() === 'done') {
+							done = true;
+						} else {
+							let first = m.content.split(/ +/)[0];
+							let regex = emojiRegex();
+							if (regex.exec(first) == null) {
+								poll_options['options'][symbols.default_emoji[options_index]] = m.content;
+							} else {
+								poll_options['options'][first] = m.content.replace(first, '').trim();
+							}
+							options_index++;
+							message.channel.messages.fetch(m.id)
+								.then(msg => msg.react('✅')).catch(e => console.log(e));
+							if (options_index == 20) {
+								done = true;
+							}
+						}
+						if (done) {
+							stage = 3;
+							message.channel.send(`Great, that's everything! Here's an example of what the poll will look like in <#${channel.id}>, if you like it, go ahead and type \`ok\` and I'll post it.`);
+							poll_options['results'] = generateResults(poll_options['options']);
+							poll_options['creator'] = {
+								'name': message.author.tag,
+								'avatar': message.author.avatarURL({size: 32})
+							};
+							let exampleEmbed = new Discord.MessageEmbed()
+								.setColor('#b22222')
+								.setTitle(`Poll: ${poll_options['name']}`)
+								.setFooter(`Poll created by ${message.author.tag}`, message.author.avatarURL({size: 32}))
+								.addField('Options (react to vote)', functions.formatOptions(poll_options['options']))
+								.addField('Results', functions.formatResults(generateTestResults(poll_options['options'])));
+							if (poll_options['description'] != '') {
+								exampleEmbed.setDescription(poll_options['description']);
+							}
+							message.channel.send(exampleEmbed);
+						}
+					} else if (stage == 3) {
+						if (m.content.toLowerCase() === 'ok') {
+							message.channel.startTyping();
+							let embed = new Discord.MessageEmbed()
+								.setColor('#b22222')
+								.setTitle(`Poll: ${poll_options['name']}`)
+								.setFooter(`Poll created by ${message.author.tag}`, message.author.avatarURL({size: 32}))
+								.addField('Options (react to vote)', functions.formatOptions(poll_options['options']))
+								.addField('Results', functions.formatResults(poll_options['results']));
+							if (poll_options['description'] != '') {
```

Comment:
I'm definitely a 80 character lines type of person, definetely helps me keep the
code from wrapping in a code review